mlflow/java/client/src/main/java/org/mlflow/artifacts/CliBasedArtifactRepository.java

```diff
@@ -223,7 +225,11 @@ private String forkMlflowProcess(List<String> mlflowCommand, String tag) {
       List<String> fullCommand = Lists.newArrayList(PYTHON_EXECUTABLE, "-m", PYTHON_COMMAND);
       fullCommand.addAll(mlflowCommand);
       ProcessBuilder pb = new ProcessBuilder(fullCommand);
-      setProcessEnvironment(pb.environment(), hostCreds);
+      if (hostCreds instanceof DatabricksMlflowHostCreds) {
+        setProcessEnvironmentDatabricks(pb.environment(), (DatabricksMlflowHostCreds) hostCreds);
+      } else {
+        setProcessEnvironment(pb.environment(), hostCreds);
+      }
```

Comment:
As an alternative approach, I considered adding a `toEnvironment()` function to the `MlflowHostCreds` interface. Extending the interface, however, would needlessly break existing third-party code which may extend this class. Further, this is the only instance in the Java Client where host credentials are mapped to environment variables + it's unclear that `toEnvironment()` meshes well with the existing `MlflowHostCreds` abstraction.

Accordingly, I think this approach is better. Feel free to chime in with pushback!